connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -340,13 +342,19 @@ public void startTask(ConnectorTaskId id,
         // Start the task before adding modifying any state, any exceptions are caught higher up the
         // call chain and there's no cleanup to do here
         workerTask.initialize(taskConfig);
+        WorkerTask existing = tasks.put(id, workerTask);
+        if (existing != null) {
+            String msg = "Task already exists in this worker; the herder should not have requested "
+                    + "that this : " + id;
```

Comment:
Yeah, lack of the Java 8 features is annoying. Since these cases indicate an incorrect usage of the `Worker` class anyway, I think we should be ok with the current handling -- at that point your worker process is in a bad state and probably needs to exit anyway.